console-subscriber/src/lib.rs

```diff
@@ -343,11 +357,40 @@ impl proto::tasks::tasks_server::Tasks for Server {
             tonic::Status::internal("cannot start new watch, aggregation task is not running")
         })?;
         let (tx, rx) = mpsc::channel(self.client_buffer);
-        permit.send(Watch(tx));
+        permit.send(WatchKind::Task(Watch(tx)));
         tracing::debug!("watch started");
         let stream = tokio_stream::wrappers::ReceiverStream::new(rx);
         Ok(tonic::Response::new(stream))
     }
+
+    async fn watch_task_details(
+        &self,
+        req: tonic::Request<proto::tasks::DetailsRequest>,
+    ) -> Result<tonic::Response<Self::WatchTaskDetailsStream>, tonic::Status> {
+        let task_id = req
+            .into_inner()
+            .id
+            .ok_or_else(|| tonic::Status::invalid_argument("missing task_id"))?;
+        let permit = self.subscribe.reserve().await.map_err(|_| {
+            tonic::Status::internal("cannot start new watch, aggregation task is not running")
+        })?;
+
+        // Check with the aggregator task to request a stream if the task exists.
+        let (stream_sender, stream_recv) = oneshot::channel();
+        permit.send(WatchKind::TaskDetail(WatchRequest {
+            id: task_id.clone(),
+            stream_sender,
+            buffer: self.client_buffer,
+        }));
+        // If the aggregator drops the sender, the task doesn't exist.
+        let rx = stream_recv
+            .await
+            .map_err(|_| tonic::Status::not_found("task not found"))?;
```

Comment:
:+1: 

(but, i might log a warning in this case, maybe in the `map_err` block?)